src/electron-main.ts

```diff
@@ -192,17 +191,22 @@ async function setupGlobals(): Promise<void> {
     }
 
     // The tray icon
-    // It's important to call `path.join` so we don't end up with the packaged asar in the final path.
-    const iconFile = `element.${process.platform === 'win32' ? 'ico' : 'png'}`;
-    iconPath = path.join(resPath, "img", iconFile);
+    // It's important to call `path.join` for the bundled assets so we don't end up with the packaged asar in the final path.
+    const iconPath = process.platform === 'win32'
+      ? (global.vectorConfig?.tray_icons?.ico || path.join(resPath, 'img', 'element.ico'))
+      : (global.vectorConfig?.tray_icons?.png || path.join(resPath, 'img', 'element.png'));
+
     global.trayConfig = {
-        icon_path: iconPath,
         brand: global.vectorConfig.brand || 'Element',
-    };
+        iconPath,
+        allowWebIconOverride: !!(
+          process.platform === 'win32' && global.vectorConfig?.tray_icons?.ico || global.vectorConfig?.tray_icons?.png
+        ),
+    } as tray.IConfig;
```

Comment:
Without this, the `IConfig` interface does not seem to actually get enforced at build (at least here anything goes).